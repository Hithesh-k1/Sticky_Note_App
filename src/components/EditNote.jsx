import React, { useState, useEffect } from "react";

const EditNote = (props) => {
  const [note, setNote] = useState(props.currentNote);

  useEffect(() => {
    setNote(props.currentNote);
  }, [props]);
  // You can tell React to skip applying an effect if certain values havenâ€™t changed between re-renders. [ props ]

  const handleInputChange = (event) => {
    const { title, value } = event.target;

    setNote({ ...note, [title]: value });
  };

  return (
    <form
      onSubmit={(event) => {
        event.preventDefault();

        props.updateNote(note.id, note);
      }}
    >
      <div key={note.id}>
        <center>
          <textarea
            placeholder="title.."
            className="form-control"
            type="text"
            title="title"
            value={note.title}
            onChange={handleInputChange}
          />
        </center>
        {/* <textarea
          placeholder="Note.."
          className="form-control"
          type="text"
          title="notes"
          value={note.notes}
          onChange={handleInputChange}
        /> */}

        <div class="form-floating">
          <textarea
            id="floatingTextarea"
            placeholder="Note.."
          className="form-control"
          type="text"
          title="notes"
          value={note.notes}
          onChange={handleInputChange}
          ></textarea>
          {/* <label for="floatingTextarea">Comments</label> */}
        </div>

        <button className="btn btn-primary">Update note</button>
        <button
          onClick={() => props.setEditing(false)}
          className="btn btn-danger"
        >
          Cancel
        </button>
      </div>
    </form>
  );
};

export default EditNote;
