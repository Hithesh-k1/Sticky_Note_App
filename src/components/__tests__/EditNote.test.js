import React from "react";
import { shallow } from "enzyme";
import EditNote from "../EditNote";

const setUp = (props = {}) => {
  const wrappedComponent = shallow(<EditNote {...props} />);
  return wrappedComponent;
};

describe("App component", () => {
  let wrappedComponent;
  let props;

  beforeEach(() => {
    const usersData = [
      { id: 1, title: "Todo 1", notes: "build CI/CD pipeline using jenkins" },
      { id: 2, title: "Todo 2", notes: "Integrate Jenkins with Sonarqube " },
      { id: 3, title: "Todo 3", notes: "Deploy to AWS S3" },
    ];

    props = {
      editing: jest.fn(),
      setEditing: jest.fn(),
      currentNote: usersData,
      updateNote: jest.fn(),
      handleInputChange:jest.fn()
    };

    wrappedComponent = setUp(props);
  });
  it("should match snapshot of EditNote Component", () => {
    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should handle setEditing event", () => {
    wrappedComponent.find("button").at(1).props().onClick();
    expect(props.setEditing).toHaveBeenCalled();
  });

  it("should handle updateNote event", () => {
    let preventDefault = jest.fn();
    const event = { preventDefault };
    wrappedComponent.find("form").props().onSubmit(event);
    expect(props.updateNote).toHaveBeenCalled();
  });

  it("should handleInputChange event", () => {
    const event = {
      target: {
        title: "test_title",
        value: "test_value",
      },
    };
    wrappedComponent.find("textarea").at(0).props().onChange(event);
    // expect(wrappedComponent.instance().state.title).toEqual("test_title");

    // expect(props.handleInputChange).toHaveBeenCalled();
  });
});
