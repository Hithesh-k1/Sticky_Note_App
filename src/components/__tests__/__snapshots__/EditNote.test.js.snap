// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot of EditNote Component 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <center>
      <textarea
        className="form-control"
        onChange={[Function]}
        placeholder="title.."
        title="title"
        type="text"
      />
    </center>
    <div
      class="form-floating"
    >
      <textarea
        className="form-control"
        id="floatingTextarea"
        onChange={[Function]}
        placeholder="Note.."
        title="notes"
        type="text"
      />
    </div>
    <button
      className="btn btn-primary"
    >
      Update note
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
